lib/dns/resource.js

```diff
@@ -880,7 +880,7 @@ function typeToClass(type) {
 function stringToClass(type) {
   assert(typeof type === 'string');
 
-  if (!hsTypes.hasOwnProperty(type))
+  if (!Object.prototype.hasOwnProperty.call(hsTypes, type))
```

Comment:
I think we want to avoid directly accessing prototypes, is your eslint config using some global rules defined elsewhere?

FWIW to avoid doing `x.hasOwnProperty(y)`, you can do: `Object.keys(x).includes(y)` for better safety